homeassistant/components/philips_js/light.py

```diff
@@ -162,46 +178,48 @@ def _calculate_effect_list(self):
         """Calculate an effect list based on current status."""
         effects = []
         effects.extend(
-            _get_effect(EFFECT_AUTO, style, setting)
+            AmbilightEffect(mode=EFFECT_AUTO, style=style, algorithm=setting)
             for style, data in self._tv.ambilight_styles.items()
-            if _is_valid(EFFECT_AUTO, style)
-            and _is_on(EFFECT_AUTO, style, self._tv.powerstate)
             for setting in data.get("menuSettings", [])
         )
 
         effects.extend(
-            _get_effect(EFFECT_EXPERT, style, algorithm)
+            AmbilightEffect(mode=EFFECT_EXPERT, style=style, algorithm=algorithm)
             for style, data in self._tv.ambilight_styles.items()
-            if _is_valid(EFFECT_EXPERT, style)
-            and _is_on(EFFECT_EXPERT, style, self._tv.powerstate)
             for algorithm in data.get("algorithms", [])
         )
 
         effects.extend(
-            _get_effect(EFFECT_MODE, style, None)
+            AmbilightEffect(mode=EFFECT_MODE, style=style)
             for style in self._tv.ambilight_modes
-            if _is_valid(EFFECT_MODE, style)
-            and _is_on(EFFECT_MODE, style, self._tv.powerstate)
         )
 
-        return sorted(effects)
+        filtered_effects = [
+            str(effect)
+            for effect in effects
+            if isinstance(effect, AmbilightEffect)
```

Comment:
```suggestion
```

Reply:
Nice one. I could just typehint the list to get autocompletion correct, doh ;-)